test/features/composer/data/html_analyzer_test.dart

```diff
@@ -0,0 +1,288 @@
+    () async {
+      const base64Data = 'iVBORw0KGgo=';
+      const htmlContent = '<img src="data:image/png;base64,$base64Data">';
+      final inlineAttachments = <String, Attachment>{};
+      final uploadUri = Uri.parse('https://example.com/upload');
+
+      when(mockUuid.v1()).thenReturn('new-uuid-123');
+      when(mockFileUploader.uploadAttachment(any, any, any))
+        .thenAnswer((_) async => Attachment(
+          blobId: Id('blob123'),
+          type: MediaType('image', 'png'),
+          disposition: ContentDisposition.inline,
+        ));
+
+      final result = await htmlAnalyzer.replaceImageBase64ToImageCID(
+        emailContent: htmlContent,
+        inlineAttachments: inlineAttachments,
+        uploadUri: uploadUri,
+      );
+
+      final document = parse(result.value1);
+      final imgTag = document.querySelector('img');
+      expect(imgTag?.attributes['src'], 'cid:new-uuid-123');
+      expect(result.value2.length, 1);
+      expect(result.value2.first.cid, 'new-uuid-123');
+      verify(mockFileUploader.uploadAttachment(any, any, uploadUri)).called(1);
+    });
+
+    test(
+      'When upload URI is null and HTML has base64 images,\n'
+      'should keep original src and not attempt uploads',
+    () async {
+      const htmlContent = '<img src="data:image/png;base64,iVBORw0KGgo=">';
+      final inlineAttachments = <String, Attachment>{};
+
+      final result = await htmlAnalyzer.replaceImageBase64ToImageCID(
+        emailContent: htmlContent,
+        inlineAttachments: inlineAttachments,
+        uploadUri: null,
+      );
+
+      expect(result.value1, htmlContent);
+      expect(result.value2, isEmpty);
+      verifyNever(mockFileUploader.uploadAttachment(any, any, any));
+    });
+
+    test(
+      'When processing multiple base64 images,\n'
+      'should replace each with a unique CID and add corresponding attachments',
+    () async {
+      const htmlContent = '''
+        <img src="data:image/png;base64,iVBORw0KGgo=">
+        <img src="data:image/jpeg;base64,/9j/4AAQSkZJRg==">
+      ''';
+      final inlineAttachments = <String, Attachment>{};
+      final uploadUri = Uri.parse('https://example.com/upload');
+
+      when(mockUuid.v1()).thenAnswer((_) => 'uuid1');
+
+      when(mockFileUploader.uploadAttachment(any, any, any))
+        .thenAnswer((_) async => Attachment(
+          blobId: Id('blob123'),
+          type: MediaType('image', 'png'),
+          disposition: ContentDisposition.inline,
+        ));
+
+      final result = await htmlAnalyzer.replaceImageBase64ToImageCID(
+        emailContent: htmlContent,
+        inlineAttachments: inlineAttachments,
+        uploadUri: uploadUri,
+      );
+
+      final document = parse(result.value1);
+      final imgTags = document.querySelectorAll('img');
+      expect(imgTags.length, 2);
+      expect(imgTags[0].attributes['src'], 'cid:uuid1');
+      expect(imgTags[1].attributes['src'], 'cid:uuid1');
```

Comment:
why 2 images have the same `src` after upload?

Reply:
In unit tests, mocking the value of Uuid.v1() only returns a value at a single point in time.

Reply:
Update expect semantics to be more accurate.